rewrite-java/src/main/java/org/openrewrite/java/ChangeMethodName.java

```diff
@@ -27,6 +28,9 @@
 @EqualsAndHashCode(callSuper = false)
 public class ChangeMethodName extends Recipe {
 
+    //A valid java method starts with a letter, an underscore or a $
+    public static final String VALID_JAVA_METHOD_PATTERN = "^[a-zA-Z_$][a-zA-Z0-9_$]*$";
```

Comment:
Did you know these special character classes?
https://github.com/openrewrite/rewrite/blob/72d79f96e281b4554f65f22ff6eb85cf8464d732/rewrite-properties/src/main/java/org/openrewrite/properties/trait/PropertiesReference.java#L63-L66

Reply:
Don't know if that's clearer here to be honest. Also notice regex can be written shorter:

```suggestion
    public static final String VALID_JAVA_METHOD_PATTERN = "^[a-zA-Z_$][\\w$]*$";
```

Reply:
I asked Tim, because I didn't know the special Character classes. He mentioned we use this in several places, so I opted for that solution.